app/src/main/java/com/willowtree/vocable/presets/adapter/PhraseAdapter.kt

```diff
@@ -6,23 +6,45 @@ import android.view.ViewGroup
 import androidx.core.view.isInvisible
 import androidx.recyclerview.widget.RecyclerView
 import com.willowtree.vocable.R
+import com.willowtree.vocable.databinding.PhraseButtonAddBinding
 import com.willowtree.vocable.databinding.PhraseButtonBinding
 import com.willowtree.vocable.room.Phrase
 import com.willowtree.vocable.utils.LocalizedResourceUtility
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 import java.util.*
 
-class PhraseAdapter(private val phrases: List<Phrase>, private val numRows: Int, private val phraseClickAction: ((Phrase) -> Unit)?) :
-    RecyclerView.Adapter<PhraseAdapter.PhraseItemViewHolder>(), KoinComponent {
+class PhraseAdapter(
+    private val phrases: List<Phrase?>,
+    private val numRows: Int,
+    private val phraseClickAction: ((Phrase) -> Unit)?,
+    private val phraseAddClickAction: (() -> Unit)?
+) :
+    RecyclerView.Adapter<PhraseAdapter.PhraseViewHolder>(), KoinComponent {
 
-    inner class PhraseItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
-        private val binding = PhraseButtonBinding.bind(itemView)
+    abstract inner class PhraseViewHolder(
+        itemView: View
+    ) : RecyclerView.ViewHolder(itemView) {
+        abstract fun bind(text: String, position: Int)
+    }
+
```

Comment:
Why do we need to create this abstract class?

Reply:
Ohh good catch, was trying another approach and gave up.

Reply:
Changing my answer here, we need that since we have 2 different ViewHolders, one for the regular button and the other for the add phrase button. They both have different views so they need different viewHolders.